Crash on start of compass

##### Detailed steps causing the problem:
- Save a cache to a list
- Open cache details
- Select default navigation icon to start the compass

##### Actual behavior after performing these steps:
- App freezes, 

##### Expected behavior after performing these steps:
- Compass should work normal


##### Version of c:geo used:
2016.11.05-NB-23c5988


##### Is the problem reproducible for you?
Yes

##### System information:
```
Samsung Galaxy S4
Android 5.0.1
```

##### Other comments and remarks:
Screenshot:
![screenshot_2016-11-05-15-17-28](https://cloud.githubusercontent.com/assets/949669/20030767/0ea7e012-a36c-11e6-89f9-967c3aecf41f.png)


Log:
```
15:21:08.680 Error cgeo 15089  [main] UncaughtException
15:21:08.680 Error cgeo 15089  java.lang.IllegalStateException: Fatal Exception thrown on Scheduler.
15:21:08.680 Error cgeo 15089  	at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:113)
15:21:08.680 Error cgeo 15089  	at android.os.Handler.handleCallback(Handler.java:739)
15:21:08.680 Error cgeo 15089  	at android.os.Handler.dispatchMessage(Handler.java:95)
15:21:08.680 Error cgeo 15089  	at android.os.Looper.loop(Looper.java:145)
15:21:08.680 Error cgeo 15089  	at android.app.ActivityThread.main(ActivityThread.java:5951)
15:21:08.680 Error cgeo 15089  	at java.lang.reflect.Method.invoke(Native Method)
15:21:08.680 Error cgeo 15089  	at java.lang.reflect.Method.invoke(Method.java:372)
15:21:08.680 Error cgeo 15089  	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1400)
15:21:08.680 Error cgeo 15089  	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1195)
15:21:08.680 Error cgeo 15089  Caused by: java.lang.IllegalArgumentException: delay < 0: -134000000
15:21:08.680 Error cgeo 15089  	at io.reactivex.android.schedulers.HandlerScheduler$HandlerWorker.schedule(HandlerScheduler.java:60)
15:21:08.680 Error cgeo 15089  	at io.reactivex.Scheduler$Worker$PeriodicTask.run(Scheduler.java:307)
15:21:08.680 Error cgeo 15089  	at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:111)
15:21:08.680 Error cgeo 15089  	... 8 more
15:21:08.680 Error cgeo 15089  [main] UncaughtException
15:21:08.680 Error cgeo 15089  java.lang.IllegalStateException: Fatal Exception thrown on Scheduler.
15:21:08.680 Error cgeo 15089  	at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:113)
15:21:08.680 Error cgeo 15089  	at android.os.Handler.handleCallback(Handler.java:739)
15:21:08.680 Error cgeo 15089  	at android.os.Handler.dispatchMessage(Handler.java:95)
15:21:08.680 Error cgeo 15089  	at android.os.Looper.loop(Looper.java:145)
15:21:08.680 Error cgeo 15089  	at android.app.ActivityThread.main(ActivityThread.java:5951)
15:21:08.680 Error cgeo 15089  	at java.lang.reflect.Method.invoke(Native Method)
15:21:08.680 Error cgeo 15089  	at java.lang.reflect.Method.invoke(Method.java:372)
15:21:08.680 Error cgeo 15089  	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1400)
15:21:08.680 Error cgeo 15089  	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1195)
15:21:08.680 Error cgeo 15089  Caused by: java.lang.IllegalArgumentException: delay < 0: -134000000
15:21:08.680 Error cgeo 15089  	at io.reactivex.android.schedulers.HandlerScheduler$HandlerWorker.schedule(HandlerScheduler.java:60)
15:21:08.680 Error cgeo 15089  	at io.reactivex.Scheduler$Worker$PeriodicTask.run(Scheduler.java:307)
15:21:08.680 Error cgeo 15089  	at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:111)
15:21:08.680 Error cgeo 15089  	... 8 more
```

--- a/main/build.gradle
+++ b/main/build.gradle
@@ -314,9 +314,8 @@ dependencies {
     compile 'com.jakewharton:process-phoenix:1.1.1'
 
     // RxJava
-    def rxjavaVersion = '2.0.0'
-    compile "io.reactivex.rxjava2:rxjava:$rxjavaVersion"
-    compile "io.reactivex.rxjava2:rxandroid:$rxjavaVersion"
+    compile "io.reactivex.rxjava2:rxjava:2.0.0"
+    compile "io.reactivex.rxjava2:rxandroid:2.0.1"
 
     // Showcase view
     // TODO replace by repository dependency. however, there is a conflict in min sdk, which needs to be solved then

--- a/main/src/cgeo/geocaching/ui/CompassView.java
+++ b/main/src/cgeo/geocaching/ui/CompassView.java
@@ -16,12 +16,8 @@ import android.view.View;
 import java.lang.ref.WeakReference;
 import java.util.concurrent.TimeUnit;
 
-import io.reactivex.BackpressureStrategy;
-import io.reactivex.Observable;
 import io.reactivex.android.schedulers.AndroidSchedulers;
 import io.reactivex.disposables.CompositeDisposable;
-import io.reactivex.functions.Consumer;
-import io.reactivex.schedulers.Schedulers;
 
 public class CompassView extends View {
 
@@ -121,16 +117,7 @@ public class CompassView extends View {
 
         initialDisplay = true;
 
-        // FIXME: this causes an exception with RxAndroid 2.0.0 (see https://github.com/cgeo/cgeo/issues/6087)
-        // periodicUpdate.add(AndroidSchedulers.mainThread().schedulePeriodicallyDirect(new UpdateAction(this), 0, 40, TimeUnit.MILLISECONDS));
-        // Use a workaround in the meantime.
-        periodicUpdate.add(Observable.interval(40, 40, TimeUnit.MILLISECONDS, Schedulers.computation()).observeOn(AndroidSchedulers.mainThread()).toFlowable(BackpressureStrategy.DROP).subscribe(new Consumer<Long>() {
-            final UpdateAction action = new UpdateAction(CompassView.this);
-            @Override
-            public void accept(final Long aLong) throws Exception {
-                action.run();
-            }
-        }));
+        periodicUpdate.add(AndroidSchedulers.mainThread().schedulePeriodicallyDirect(new UpdateAction(this), 0, 40, TimeUnit.MILLISECONDS));
     }
 
     @Override

